File: windows/Liferay.SDK/Service/V73/Cpoptioncategory/CpoptioncategoryService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Cpoptioncategory
{
	public class CpoptioncategoryService : ServiceBase
	{
		public CpoptioncategoryService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> GetCpOptionCategoryAsync(long cpOptionCategoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpOptionCategoryId", cpOptionCategoryId);

			var _command = new JsonObject()
			{
				{ "/commerce.cpoptioncategory/get-cp-option-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCpOptionCategoryAsync(IDictionary<string, string> titleMap, IDictionary<string, string> descriptionMap, double priority, string key, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("titleMap", titleMap);
			_parameters.Add("descriptionMap", descriptionMap);
			_parameters.Add("priority", priority);
			_parameters.Add("key", key);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.cpoptioncategory/add-cp-option-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCpOptionCategoryAsync(long cpOptionCategoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpOptionCategoryId", cpOptionCategoryId);

			var _command = new JsonObject()
			{
				{ "/commerce.cpoptioncategory/delete-cp-option-category", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> FetchCpOptionCategoryAsync(long cpOptionCategoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpOptionCategoryId", cpOptionCategoryId);

			var _command = new JsonObject()
			{
				{ "/commerce.cpoptioncategory/fetch-cp-option-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> SearchCpOptionCategoriesAsync(long companyId, string keywords, int start, int end, JsonObjectWrapper sort)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("keywords", keywords);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "sort", "$languageUtil.getJSONWrapperClassName($parameter.type)", sort);

			var _command = new JsonObject()
			{
				{ "/commerce.cpoptioncategory/search-cp-option-categories", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCpOptionCategoryAsync(long cpOptionCategoryId, IDictionary<string, string> titleMap, IDictionary<string, string> descriptionMap, double priority, string key)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpOptionCategoryId", cpOptionCategoryId);
			_parameters.Add("titleMap", titleMap);
			_parameters.Add("descriptionMap", descriptionMap);
			_parameters.Add("priority", priority);
			_parameters.Add("key", key);

			var _command = new JsonObject()
			{
				{ "/commerce.cpoptioncategory/update-cp-option-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/cpoptioncategory/CpoptioncategoryService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.cpoptioncategory;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CpoptioncategoryService extends BaseService {

	public CpoptioncategoryService(Session session) {
		super(session);
	}

	public JSONObject addCpOptionCategory(JSONObject titleMap, JSONObject descriptionMap, double priority, String key, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("descriptionMap", checkNull(descriptionMap));
			_params.put("key", checkNull(key));
			_params.put("priority", priority);
			_params.put("titleMap", checkNull(titleMap));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.cpoptioncategory/add-cp-option-category", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCpOptionCategory(long cpOptionCategoryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpOptionCategoryId", cpOptionCategoryId);

			_command.put("/commerce.cpoptioncategory/delete-cp-option-category", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject fetchCpOptionCategory(long cpOptionCategoryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpOptionCategoryId", cpOptionCategoryId);

			_command.put("/commerce.cpoptioncategory/fetch-cp-option-category", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getCpOptionCategory(long cpOptionCategoryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpOptionCategoryId", cpOptionCategoryId);

			_command.put("/commerce.cpoptioncategory/get-cp-option-category", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject searchCpOptionCategories(long companyId, String keywords, int start, int end, JSONObjectWrapper sort) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("keywords", checkNull(keywords));
			_params.put("start", start);
			mangleWrapper(_params, "sort", "com.liferay.portal.kernel.search.Sort", sort);

			_command.put("/commerce.cpoptioncategory/search-cp-option-categories", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCpOptionCategory(long cpOptionCategoryId, JSONObject titleMap, JSONObject descriptionMap, double priority, String key) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpOptionCategoryId", cpOptionCategoryId);
			_params.put("descriptionMap", checkNull(descriptionMap));
			_params.put("key", checkNull(key));
			_params.put("priority", priority);
			_params.put("titleMap", checkNull(titleMap));

			_command.put("/commerce.cpoptioncategory/update-cp-option-category", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}